shurunxuan/IGME750-DSEngine
# System/audio

To test:
 * Set `DSEngineTestProject` as the startup project.
 * Download the latest assets pack from Google Drive.
 * Compile and run.
 * Use `z` to play the first audio source. Use `x` to pause. Use `c` to stop.
 * Use `1` to make the pitch of the first audio source higher. Use `2` to make the pitch lower. Observe the speed of the audio will also be changed. This is because the time streching / pitch shifting is not implemeted yet.
 * Use `3` to make the volume of the first audio source higher. Use `4` to make the volume lower.
 * Use `v` to play the second audio source. Use `b` to pause. Use `n` to stop.
 * Move the camera with `w`, `a`, `s` and `d` and mouse pointer. Observe the difference between the behaviors of the two sources.
 * Go to line 74~80 of file `DSEngineTestProject/TestGameApp.cpp`.
 * Observe how to set an audio source as looping, 3D, and the Doppler Scaler and Curve Distance Scaler.
 * Go to line 85~86 of file `DSEngineTestProject/TestGameApp.cpp`.
 * Observe how to load an audio file.
 * Go to `PressSpaceToPlayAudio::Update` in file `DSEngineTestProject/PressSpaceToPlayAudio.cpp`.
 * Observe how to play, pause, stop audio playback, as well as adjusting the pitch and volume.
